tests/DummyServerLib/DummyServer.cpp

```diff
@@ -89,17 +96,21 @@ void DummyServer::deactivate() {
     } while(!_mainLoopThread.try_join_for(boost::chrono::milliseconds(10)));
   }
 
-  // Remove the SerialPort which is accessing the virtual serial ports provided by socat
+  // Remove the SerialPort which is accessing the virtual serial ports provided by socat.
   _serialPort.reset();
 
-  // Finally, stop the socat runner
-  std::cout << "DEBUG!!! << DummyServer::deactivate() stopping socat runner." << std::endl;
+  // Finally, stop the socat runner.
+  if(_debug) {
+    std::cout << "DEBUG!!! << DummyServer::deactivate() stopping socat runner." << std::endl;
+  }
   assert(_socatRunner.running());
   _socatRunner.terminate();
   _socatRunner.wait();
 
-  // Wait for "front door" file descriptor to become invalid
-  std::cout << "DEBUG!!! << DummyServer::deactivate() waiting for front door to close." << std::endl;
+  // Wait for the "front door" file descriptor to become invalid.
+  if(_debug) {
+    std::cout << "DEBUG!!! << DummyServer::deactivate() waiting for front door to close." << std::endl;
```

Comment:
This one is part of the hack of understanding the race conditions in the unified test. It has to stay active even if _debug is false. We will remove this before the release.